src/rpc/server.rs

```diff
@@ -289,18 +292,46 @@ async fn execute_tx_impl(
     }
 
     match gas_station
-        .execute_transaction(reservation_id, tx_data, user_sig)
+        .execute_transaction(reservation_id, tx_data, user_sig, options.clone())
         .await
     {
-        Ok(effects) => {
+        Ok(mut tx_block_response) => {
+            if tx_block_response.effects.is_none() {
+                error!("Failed to execute transaction: Missing transaction effects");
+                metrics.num_failed_execute_tx_requests.inc();
+                return (
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                    Json(ExecuteTxResponse::new_err(anyhow::anyhow!(
+                        "Transaction execution failed: no effects returned"
+                    ))),
+                );
+            }
             info!(
                 ?reservation_id,
                 "Successfully executed transaction {:?} with status: {:?}",
-                effects.transaction_digest(),
-                effects.status()
+                tx_block_response
```

Comment:
There's two unwraps here. I think it would be better to do a 
```suggestion
                if let Some(ref effects) = tx_block_response.effects {
                 ...
                } else {
                  error!("Failed to execute transaction: Missing transaction effects");
                  metrics.num_failed_execute_tx_requests.inc();
                  
                    return (
                        StatusCode::INTERNAL_SERVER_ERROR,
                        Json(ExecuteTxResponse::new_err(anyhow::anyhow!(
                            "Transaction execution failed: no effects returned"
                         ))),
                         );
                }
```

Reply:
oh yes thats much nicer, thanks Stefan! 🙏 It starts getting a bit too nested but its still readable 👍